umpost/conversion_driver_esm1p5.py

```diff
@@ -249,15 +315,20 @@ def convert_esm1p5_output_dir(esm1p5_output_dir):
         warnings.warn(
             f"No files matching pattern '{fields_file_name_pattern}' "
             f"found in {current_atm_output_dir.resolve()}. No files will be "
-            "converted to NetCDF."
+            "converted to netCDF."
         )
 
         return [], []  # Don't try to run the conversion
 
-    succeeded, failed = convert_fields_file_list(
-        atm_dir_fields_files,
-        nc_write_dir
-    )
+    input_output_pairs = [
+        (
+            ff_path,
+            get_nc_write_path(ff_path, nc_write_dir, get_ff_date(ff_path))
+        )
+        for ff_path in atm_dir_fields_files
+    ]
```

Comment:
We can rework this multi-line comprehension into 2 steps:
```suggestion
    outputs = [get_nc_write_path(path, nc_write_dir, get_ff_date(path)) for path in atm_dir_fields_files]
    input_output_pairs = zip(atm_dir_fields_files, outputs)
```
I find this clearer as the output path & in/out tuple concatenation steps are separate. Also, less brackets are required, which were getting forced onto separate lines by the formatter.

Reply:
Added in 797a824

Just to understand better, is the main reason to use the list comprehension over the manual `append` that it's faster? Probably because I'm still newish to Python I still do find the manual loop + `append` the easiest to read.

Reply:
Comprehensions should be faster, although they're the recommended language idiom for creating data sequences. The gist is they're more declarative than the imperative style of manually assembling new data structures in a stepwise fashion.
This talk should help explain core idioms & describe data interfaces for composing code, _Transforming Code Into Beautiful, Idiomatic Python | Raymond Hettinger_:
https://www.youtube.com/watch?v=anrOzOapJ2E
List comprehensions are around the 46 minute mark, but early on he discusses moving away from manual looping.
(The speaker is a core Python developer)

Reply:
Awesome, thanks!